frontend/templates/product/sections/ProductSection/ProductOptions.tsx

```diff
@@ -113,19 +111,60 @@ enum SelectorType {
    IMAGE_RADIO,
 }
 
-type Variant = Product['variants'][0];
-
 type Option = Product['options'][0];
 
 function findVariant(
-   variants: Variant[],
-   selectedOptions: Record<string, string>
+   product: Product,
+   previousOptions: Record<string, string>,
+   optionsUpdate: Record<string, string>
 ) {
-   return variants.find((variant) => {
+   const { variants, options } = product;
+   const newOptions = { ...previousOptions, ...optionsUpdate };
+
+   const exactMatch = variants.find((variant) => {
       return variant.selectedOptions.every((option) => {
-         return selectedOptions[option.name] === option.value;
+         return newOptions[option.name] === option.value;
+      });
+   });
+
+   if (exactMatch) {
+      return exactMatch;
+   }
+
+   const productOptionsNames = options.map((option) => option.name);
+   const optionsUpdateNames = Object.keys(optionsUpdate);
+
+   const variantsMatchingUpdate = variants.filter((variant) => {
+      return optionsUpdateNames.every((name) => {
+         const correspondingVariantOption = variant.selectedOptions.find(
+            (option) => option.name === name
+         );
+         return correspondingVariantOption?.value === optionsUpdate[name];
       });
```

Comment:
With only one option update, this could be simplified to

```suggestion
      return optionUpdate.value === variant.selectedOptions.find(
         (option) => option.name === optionUpdate.name
      )?.value;
```

Reply:
See answer above